drivers/timer/esp32c3_sys_timer.c

```diff
@@ -134,3 +134,15 @@ uint32_t sys_clock_cycle_get_32(void)
 {
 	return systimer_ll_get_counter_value_low(SYSTIMER_COUNTER_1);
 }
+
+uint64_t sys_clock_cycle_get_64(void)
+{
+	k_spinlock_key_t key = k_spin_lock(&lock);
+	uint64_t ret = (uint64_t)systimer_ll_get_counter_value_high(SYSTIMER_COUNTER_1) << 32;
+
+	ret |= systimer_ll_get_counter_value_low(SYSTIMER_COUNTER_1);
+
+	k_spin_unlock(&lock, key);
+
```

Comment:
Common scheme is to read the high value again after reading the lower value.
see ./include/arch/arm/aarch32/cortex_a_r/timer.h
```
	uint32_t lower;
	uint32_t upper, upper_saved;

	/* To get the value from the Global Timer Counter register proceed
	 * as follows:
	 * 1. Read the upper 32-bit timer counter register.
	 * 2. Read the lower 32-bit timer counter register.
	 * 3. Read the upper 32-bit timer counter register again. If the value
	 * is different to the 32-bit upper value read previously,
	 * go back to step 2.
	 * Otherwise the 64-bit timer counter value is correct.
	 */
	upper = sys_read32(TIMER_REG_GET(TIMER_CNT_UPPER));
	do {
		upper_saved = upper;
		lower = sys_read32(TIMER_REG_GET(TIMER_CNT_LOWER));
		upper = sys_read32(TIMER_REG_GET(TIMER_CNT_UPPER));
	} while (upper != upper_saved);

	return ((uint64_t)upper) << 32 | lower;